### ISSUE TITLE
Inconsistent Liquidations in BaseStrategy Emergency Exit Mode

### ISSUE THREAD

**Issue by fubuloubu** [yearn/yearn-vaults#229]
1. We are in emergency mode and are starting a `harvest`:

```solidity
function harvest() external onlyKeepers {
    uint256 profit = 0;
    uint256 loss = 0;
    uint256 debtOutstanding = vault.debtOutstanding();
```

2. The vault will return `strategy_totalDebt`:

```py
def _debtOutstanding(strategy: address) -> uint256:
    # See note on `debtOutstanding()`.
    strategy_debtLimit: uint256 = self.strategies[strategy].debtRatio * self._totalAssets() / MAX_BPS
    strategy_totalDebt: uint256 = self.strategies[strategy].totalDebt
    ...
    else:
        return strategy_totalDebt - strategy_debtLimit
```
For simplicity we assume debtOutstanding = strategy_totalDebt = 100

3. We go through this part of harvest:

```solidity
if (emergencyExit) {
    // Free up as much capital as possible
    uint256 totalAssets = estimatedTotalAssets();
    // NOTE: use the larger of total assets or debt outstanding to book losses properly
    (debtPayment, loss) = liquidatePosition(totalAssets > debtOutstanding ? totalAssets : debtOutstanding);
    // NOTE: take up any remainder here as profit
    if (debtPayment > debtOutstanding) {
        profit = debtPayment.sub(debtOutstanding);
        debtPayment = debtOutstanding;
    }
    ...
```
We assume that totalAssets = 110.

Then we will call `liquidatePosition(110)`.

We assume that the return values are: `debtPayment` = 105, `loss` = 5

Then `profit` = 5, `debtPayment` = 100.

4. Strategy calls `report`:

```solidity
        debtOutstanding = vault.report(profit, loss, debtPayment);
```
The values are: 5, 5, 100

5. We enter `_reportLoss`:

```py
    # Loss can only be up the amount of debt issued to strategy
    totalDebt: uint256 = self.strategies[strategy].totalDebt
    assert totalDebt >= loss
    self.strategies[strategy].totalLoss += loss
    self.strategies[strategy].totalDebt = totalDebt - loss
    self.totalDebt -= loss
```

Hence, strategy_totaldebt = 95.

6. Now we handle the `debtPayment`:

```py
    # Outstanding debt the Strategy wants to take back from the Vault (if any)
    # NOTE: debtOutstanding <= StrategyParams.totalDebt
    debt: uint256 = self._debtOutstanding(msg.sender)
    debtPayment: uint256 = min(_debtPayment, debt)

    if debtPayment > 0:
        self.strategies[msg.sender].totalDebt -= debtPayment
        self.totalDebt -= debtPayment
        debt -= debtPayment
        # NOTE: `debt` is being tracked for later
```

`debt` = 95 (just has been updated)

`debtPayment` = min(95, 100) = 95

`strategy_totalDebt` = 95 - 95 = 0

`debt` = 95 - 95 = 0

7. The `transfer` takes place:

```py
    totalAvail: uint256 = gain + debtPayment
    if totalAvail < credit:  # credit surplus, give to Strategy
        self.erc20_safe_transfer(self.token.address, msg.sender, credit - totalAvail)
```
totalAvail = 5 + 95 = 100

Hence 100 will be transferred, but 105 would have been available.

What we didn't realize it that a subsequent call to `harvest` should transfer the remaining 5 also.

---

Solution is to refactor `liquidatePosition(totalAssets > debtOutstanding ? totalAssets : debtOutstanding);` to instead be `liquidateEverything(debtOutstanding);`
